api/v2/views/machine_request.py

```diff
@@ -62,11 +64,23 @@ def _get_new_provider(self):
             raise Exception("settings.REPLICATION_PROVIDER_LOCATION could not be set. Contact a developer.")
 
 
+    def filter_tags(self, user, tag_names):
+        new_tag_names = []
+        for tag in sorted(tag_names.split(", ")):
+            core_tag = Tag.objects.filter(name__iexact=tag).first()
+            if not core_tag:
+                continue
+            if not core_tag.allow_access(user):
+                continue
+            new_tag_names.append(tag)
+        return ", ".join(new_tag_names)
+
     def perform_create(self, serializer):
 
+        request_user = self.request.user
```

Comment:
Wouldn't this need to be an instance field to be available in the `get_queryset()`? Or magic ✨ ?